src/GraphQL/Mutations/SetSession.php

```diff
@@ -40,4 +42,40 @@ public function setCurrency($rootValue, array $args, GraphQLContext $context, Re
 
         return $sessionService->get();
     }
+
+    // setGeoLocation
+    public function setGeoLocation($rootValue, array $args, GraphQLContext $context, ResolveInfo $resolveInfo)
+    {
+        $ip_address = $args['input']['ip_address'];
+
+        if (!isset($ip_address)) {
```

Comment:
@AlexeiaW for this part, we don't really want to give the user an opportunity to input an IP, rather we can just pull this value from $_SERVER['REMOTE_ADDR']

For testing we can use this phpunit feature to mock the IP address:

https://stackoverflow.com/a/53283002/558287

Technically speaking, we may want to run this method from inside the initial session token creation function, but this is a nice placeholder for this functionality for now. 

(This feature could be used to restrict certain shopping items to regions, so we wouldn't want the user to be able to set this manually in anyway, and rely on the IP reporting from their connection)

Reply:
Yeah it makes sense, we wouldn't want there be a way for external clients to change the ip address, or provide a fake one